aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/ArnUtils.java

```diff
@@ -0,0 +1,48 @@
+package software.amazon.codeartifact.domain;
+
+public class ArnUtils {
+    public static final String SERVICE_NAME = "codeartifact";
+    public static final String DOMAIN_RESOURCE_TYPE = "domain";
+
+    public static DomainArn fromArn(String arn) {
+
+        // TODO (jonjara) move ArnUtils to a common module
+        final int NUMBER_OF_ARN_FIELDS = 6;
+        // Sample ARNs:
+        // arn:aws:codeartifact:<region>:<domain-owner>:domain/<domain-name>
+        // arn:aws:codeartifact:<region>:<domain-owner>:repository/<domain-name>/<repo-name>
```

Comment:
you have a repo example here, but this is a `DomainArn` why not just make this an `Arn` class that handles both?

Reply:
ok will generalize and move later.